uktrade/data-hub-frontend
# Fix logic to show project value if either date before cut off

## Description of change

This fix follows a recent PR which added logic to only display the Project Value field for an investment project if the project's actual or estimated land date was before 01/04/2020. 

There is a bug in that when the estimated land date is after 01/04/2020 but the actual land date is before, this expression returns false: `Date(estimatedLandDate || actualLandDate) < valueCutOffDate`.

This PR fixes the logic in the editValue controller by first eliminating the null fields (because 
`new Date(null)` returns 01/01/1970) and then filtering the results by whether they're before the cut off date. 

I have also added more test cases to cover all possible combinations of estimated and actual land dates.

## Test instructions

Go to an investment project.
Change the estimated land date to after 01/04/200.
Change the actual land date to before 01/04/2020.
Click Add/Edit Value on the project details page.
Check the Project Value field appears.


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
